src/__tests__/use-script.test.tsx

```diff
@@ -57,10 +57,51 @@ describe('useScript', () => {
     it('should render a script only once', () => {
         expect(document.querySelectorAll('script').length).toBe(0);
 
-        renderHook(() => useScript({ src: 'http://scriptsrc/' }));
+        const props = { src: 'http://scriptsrc/' };
+        const handle = renderHook((p) => useScript(p), {
+            initialProps: props,
+        });
         expect(document.querySelectorAll('script').length).toBe(1);
 
-        renderHook(() => useScript({ src: 'http://scriptsrc/' }));
+        handle.rerender();
```

Comment:
The original test was equivalent to testing two separate components. This test is more akin to how the library is used.

If it was desired to avoid duplicate tags across components, I believe external state would be the correct solution in any case.